File: Colaveco/JavaModel/mobile/Android/gxcommon/com/colaveco/pdfdownload_level_detail.java
/*
               File: PDFdownload_Level_Detail
        Description: PDFdownload_Level_Detail
             Author: GeneXus Android Generator version 17_0_7-156546
       Generated on: November 14, 2022 8:52:6.94
       Program type: Callable routine
          Main DBMS: SQLite
*/
package com.colaveco ;
import com.colaveco.*;
import com.genexus.*;
import com.genexus.db.*;
import java.sql.*;
import com.artech.base.services.*;

public final  class pdfdownload_level_detail extends GXProcedure implements IGxProcedure
{
   public boolean execute( IPropertiesObject androidPropertiesObject )
   {

      String aP0 = "";
      long aP1 = 0;
      int aP2 = 0;
      com.colaveco.SdtPDFdownload_Level_DetailSdt[] aP3 = new com.colaveco.SdtPDFdownload_Level_DetailSdt[] {new com.colaveco.SdtPDFdownload_Level_DetailSdt()};

      aP0 = (String) androidPropertiesObject.optStringProperty("resultadoPathPdf");
      aP1 = (long) GXutil.lval( androidPropertiesObject.optStringProperty("resultadoFicha"));
      aP2 = (int) GXutil.lval( androidPropertiesObject.optStringProperty("gxid"));

      execute(aP0, aP1, aP2, aP3);

      IEntity outObjGXM1PDFdownload_Level_DetailSdt =  AndroidContext.ApplicationContext.createEntity("", "PDFdownload_Level_Detail", null);
      if (aP3[0] != null)
      {
         aP3[0].sdttoentity(outObjGXM1PDFdownload_Level_DetailSdt );
      }
      androidPropertiesObject.setProperty("Gx_Output",outObjGXM1PDFdownload_Level_DetailSdt);


      return true ;
   }

   public pdfdownload_level_detail( int remoteHandle )
   {
      super( remoteHandle , new ModelContext( pdfdownload_level_detail.class ), "" );
   }

   public pdfdownload_level_detail( int remoteHandle ,
                                    ModelContext context )
   {
      super( remoteHandle , context, "" );
   }

   public com.colaveco.SdtPDFdownload_Level_DetailSdt executeUdp( String aP0 ,
                                                                  long aP1 ,
                                                                  int aP2 )
   {
      pdfdownload_level_detail.this.AV5resultadoPathPdf = aP0;
      pdfdownload_level_detail.this.AV6resultadoFicha = aP1;
      pdfdownload_level_detail.this.AV13gxid = aP2;
      pdfdownload_level_detail.this.aP3 = new com.colaveco.SdtPDFdownload_Level_DetailSdt[] {new com.colaveco.SdtPDFdownload_Level_DetailSdt()};
      initialize();
      privateExecute();
      return aP3[0];
   }

   public void execute( String aP0 ,
                        long aP1 ,
                        int aP2 ,
                        com.colaveco.SdtPDFdownload_Level_DetailSdt[] aP3 )
   {
      execute_int(aP0, aP1, aP2, aP3);
   }

   private void execute_int( String aP0 ,
                             long aP1 ,
                             int aP2 ,
                             com.colaveco.SdtPDFdownload_Level_DetailSdt[] aP3 )
   {
      pdfdownload_level_detail.this.AV5resultadoPathPdf = aP0;
      pdfdownload_level_detail.this.AV6resultadoFicha = aP1;
      pdfdownload_level_detail.this.AV13gxid = aP2;
      pdfdownload_level_detail.this.aP3 = aP3;
      initialize();
      /* GeneXus formulas */
      /* Output device settings */
      privateExecute();
   }

   private void privateExecute( )
   {
      Gxids = "gxid_" + GXutil.str( AV13gxid, 8, 0) ;
      if ( GXutil.strcmp(Gxwebsession.getValue(Gxids), "") == 0 )
      {
         AV7url = "https://www.colaveco.com.uy/gestor/utils/bajar_archivo.php?archivo=" + AV5resultadoPathPdf ;
         new com.colaveco.loger(remoteHandle, context).execute( AV7url) ;
         AV8path = com.genexus.util.GXDirectory.applicationdatapath() + AV9file.getSeparator() + GXutil.trim( GXutil.str( AV6resultadoFicha, 18, 0)) + ".pdf" ;
         new com.colaveco.loger(remoteHandle, context).execute( AV8path) ;
         AV9file.setSource( AV8path );
         AV12httpClient.execute("GET", AV7url);
         AV12httpClient.toFile(AV8path);
         AV10uri = "file://" + AV8path ;
         new com.colaveco.loger(remoteHandle, context).execute( AV10uri) ;
         AV11destfile = AV10uri ;
         Gxwebsession.setValue(Gxids+"gxvar_Destfile", AV11destfile);
         Gxwebsession.setValue(Gxids, "true");
      }
      else
      {
         AV11destfile = Gxwebsession.getValue(Gxids+"gxvar_Destfile") ;
      }
      AV16GXM1PDFdownload_Level_DetailSdt.setgxTv_SdtPDFdownload_Level_DetailSdt_Destfile( AV11destfile );
      cleanup();
   }

   protected void cleanup( )
   {
      this.aP3[0] = pdfdownload_level_detail.this.AV16GXM1PDFdownload_Level_DetailSdt;
      CloseOpenCursors();
      exitApplication();
   }

   protected void CloseOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public void initialize( )
   {
      AV16GXM1PDFdownload_Level_DetailSdt = new com.colaveco.SdtPDFdownload_Level_DetailSdt(remoteHandle, context);
      Gxids = "" ;
      Gxwebsession = AndroidContext.ApplicationContext.getAndroidSession();
      AV7url = "" ;
      AV8path = "" ;
      AV9file = new com.genexus.util.GXFile();
      AV12httpClient = new com.genexus.internet.HttpClient();
      AV10uri = "" ;
      AV11destfile = "" ;
      /* GeneXus formulas. */
      Gx_err = (short)(0) ;
   }

   private short Gx_err ;
   private int AV13gxid ;
   private long AV6resultadoFicha ;
   private String Gxids ;
   private String AV5resultadoPathPdf ;
   private String AV7url ;
   private String AV8path ;
   private String AV10uri ;
   private String AV11destfile ;
   private com.genexus.util.GXFile AV9file ;
   private com.colaveco.SdtPDFdownload_Level_DetailSdt[] aP3 ;
   private com.genexus.internet.HttpClient AV12httpClient ;
   private com.artech.base.services.IAndroidSession Gxwebsession ;
   private com.colaveco.SdtPDFdownload_Level_DetailSdt AV16GXM1PDFdownload_Level_DetailSdt ;
}

File: Colaveco/JavaModel/mobile/iOS/gxcommon/pdfdownload_level_detail.swift
/*
               File: PDFdownload_Level_Detail
        Description: PDFdownload_Level_Detail
             Author: GeneXus Swift Generator version 17_0_7-156546
       Generated on: November 14, 2022 8:52:17.20
       Program type: Callable routine
          Main DBMS: SQLite
*/

@objc( pdfdownload_level_detail )
public final  class pdfdownload_level_detail: GXProcedure
{
   @objc public func executeFlexibleClient( _ aP0: NSString? ,
                                            _ aP1: NSNumber? ,
                                            _ aP2: NSNumber? ,
                                            _ inout_aP3: AutoreleasingUnsafeMutablePointer<SdtPDFdownload_Level_DetailSdt?> )
   {
      let aP3: SdtPDFdownload_Level_DetailSdt? = inout_aP3.pointee
      let iP0: String = (aP0 == nil) ? "" : aP0! as String
      let iP1: Int64 = (aP1 == nil) ? 0 : aP1!.int64Value as Int64
      let iP2: Int = (aP2 == nil) ? 0 : aP2!.intValue as Int
      var iP3: SdtPDFdownload_Level_DetailSdt = (aP3 == nil) ? SdtPDFdownload_Level_DetailSdt() : aP3! as SdtPDFdownload_Level_DetailSdt
      execute(iP0, iP1, iP2, &iP3)
      inout_aP3.pointee  = iP3 as SdtPDFdownload_Level_DetailSdt?
   }

   public func executeUdp( _ aP0: String ,
                           _ aP1: Int64 ,
                           _ aP2: Int ) -> SdtPDFdownload_Level_DetailSdt
   {
      self.AV5resultadoPathPdf = aP0
      self.AV6resultadoFicha = aP1
      self.AV13gxid = aP2
      initialize()
      autoreleasepool
      {
         privateExecute()
      }
      return self.AV16GXM1PDFdownload_Level_DetailSdt
   }

   public func execute( _ aP0: String ,
                        _ aP1: Int64 ,
                        _ aP2: Int ,
                        _ aP3: inout SdtPDFdownload_Level_DetailSdt )
   {
      execute_int(aP0, aP1, aP2, &aP3)
   }

   private func execute_int( _ aP0: String ,
                             _ aP1: Int64 ,
                             _ aP2: Int ,
                             _ aP3: inout SdtPDFdownload_Level_DetailSdt )
   {
      self.AV5resultadoPathPdf = aP0
      self.AV6resultadoFicha = aP1
      self.AV13gxid = aP2
      initialize()
      /* GeneXus formulas */
      /* Output device settings */
      autoreleasepool
      {
         privateExecute()
      }
      aP3 = self.AV16GXM1PDFdownload_Level_DetailSdt
   }

   private func privateExecute( )
   {
      Gxids = "gxid_"  +  String.gxStringFromNumber(Decimal(AV13gxid), totalLength:8, decimals:0)
      if ( GXutil.strcmp(Gxwebsession.getValue(Gxids), "") == 0 )
      {
         AV7url = "https://www.colaveco.com.uy/gestor/utils/bajar_archivo.php?archivo="  +  AV5resultadoPathPdf
         loger().execute( AV7url)
         AV8path = GXDirectory.applicationdatapath()  +  AV9file.separator  +  String.gxStringFromNumber(Decimal(AV6resultadoFicha), totalLength:18, decimals:0).gxStringByTrimming()  +  ".pdf"
         loger().execute( AV8path)
         AV9file.source = AV8path
         AV12httpClient.execute("GET", url:AV7url)
         AV12httpClient.toFile(AV8path)
         AV10uri = "file://"  +  AV8path
         loger().execute( AV10uri)
         AV11destfile = AV10uri
         Gxwebsession.setValue(Gxids+"gxvar_Destfile", AV11destfile)
         Gxwebsession.setValue(Gxids, "true")
      }
      else
      {
         AV11destfile = Gxwebsession.getValue(Gxids+"gxvar_Destfile")
      }
      AV16GXM1PDFdownload_Level_DetailSdt.gxTv_SdtPDFdownload_Level_DetailSdt_Destfile = AV11destfile
      cleanup()
   }

   public override func cleanup( )
   {
      closeOpenCursors()
   }

   private func closeOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public override func initialize( )
   {
      AV16GXM1PDFdownload_Level_DetailSdt = SdtPDFdownload_Level_DetailSdt()
      Gxids = ""
      Gxwebsession = GXOfflineSession.sharedInstance()
      AV7url = ""
      AV8path = ""
      AV9file = GXFile()
      AV12httpClient = GXHttpClient()
      AV10uri = ""
      AV11destfile = ""
      /* GeneXus formulas. */
      Gx_err = 0
   }

   private var AV13gxid : Int = 0
   private var Gx_err : Int = 0
   private var AV6resultadoFicha : Int64 = 0
   private var Gxids : String = ""
   private var AV5resultadoPathPdf : String = ""
   private var AV7url : String = ""
   private var AV8path : String = ""
   private var AV10uri : String = ""
   private var AV11destfile : String = ""
   private var AV9file : GXFile = GXFile()
   private var AV12httpClient : GXHttpClient = GXHttpClient()
   private var Gxwebsession : GXOfflineSession = GXOfflineSession()
   private var AV16GXM1PDFdownload_Level_DetailSdt : SdtPDFdownload_Level_DetailSdt = SdtPDFdownload_Level_DetailSdt()
}

